### PR TITLE
Scale test scripts updates to support Terasort benchmark, improved S3 write time

### PR DESCRIPTION
**Pull request by akirillov**
### What changes were proposed in this pull request?
Resolves [DCOS-60318: Improve gensort job to avoid high latencies while writing to S3](https://jira.mesosphere.com/browse/DCOS-60318)

Originally started as gensort write time improvement which is achieved via committer configuration property:
```
"spark.hadoop.mapreduce.fileoutputcommitter.algorithm.version": "2"
```
This PR includes refactoring to separate scale tests from the sort benchmark and introduces changes to support ongoing work in https://github.com/mesosphere/spark-build/pull/559 and https://github.com/mesosphere/spark-build/pull/560

### Why are the changes needed?
These changes are needed to support new scale tests objectives and run the workloads separately.

### How were the changes tested?
* by running `gensort` job on a cluster with 40 executors and verifying 1TiB dataset is generated in under 1 hour
* by running all the scripts against private cluster to test changes from https://github.com/mesosphere/spark-build/pull/559 and https://github.com/mesosphere/spark-build/pull/560

### CODE REVIEW

File: scale-tests/scripts/terasort.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+. ${SCRIPT_DIR}/aws_credentials.sh
+
+cat ${TEMPLATES_DIR}/sort-application.tmpl \
+  | sed "s|AWS_ACCESS_KEY_ID|${AWS_ACCESS_KEY_ID:-}|" \
+  | sed "s|AWS_SECRET_ACCESS_KEY|${AWS_SECRET_ACCESS_KEY:-}|" \
+  | sed "s|AWS_SESSION_TOKEN|${AWS_SESSION_TOKEN:-}|" \
+  | sed "s|S3_ENDPOINT|${S3_ENDPOINT}|" \
+  | sed "s|NUM_EXECUTORS|${NUM_EXECUTORS}|" \
+  | sed "s|SOURCE_PATH|${2:-}|" \
+  | sed "s|TARGET_PATH|${3:-}|" | kubectl --namespace "${NAMESPACE}" apply -f -
```
Comment (VladimirKravtsov) [inline scale-tests/scripts/terasort.sh]: Here is a missed new line and maybe move the `| kubectl --namespace "${NAMESPACE}" apply -f -` on new line would be better?
```suggestion
  | sed "s|TARGET_PATH|${3:-}|" 
  | kubectl --namespace "${NAMESPACE}" apply -f -
  
```

File: scale-tests/templates/gensort-application.tmpl [hunk 2]
```diff
@@ -39,7 +40,7 @@ spec:
     memory: "1g"
     labels:
       version: 2.4.3
-    serviceAccount: spark-driver
+    serviceAccount: spark-service-account
```
Comment (farhan5900) [inline scale-tests/templates/gensort-application.tmpl]: Should service account name come from `gensort.sh` using environment variable? As user can change it by exporting `SERVICE_ACCOUNT_NAME` environment variable.
```suggestion
    serviceAccount: SERVICE_ACCOUNT_NAME
```

File: scale-tests/templates/sort-application.tmpl [hunk 3]
```diff
@@ -1,39 +1,43 @@
...
     type: Never
   driver:
     cores: 2
-    memory: "2g"
+    memory: "6g"
     labels:
       version: 2.4.3
       metrics-exposed: "true"
-    serviceAccount: spark-driver
+    serviceAccount: spark-service-account
```
Comment (farhan5900) [inline scale-tests/templates/sort-application.tmpl]: Similar as `gensort` template.

File: scale-tests/scripts/terasort.sh (hunk 1 above)
Comment (akirillov) [reply scale-tests/scripts/terasort.sh]: sounds good, thanks for pointing that out. updated here and in other similar scripts

File: scale-tests/templates/gensort-application.tmpl (hunk 2 above)
Comment (akirillov) [reply scale-tests/templates/gensort-application.tmpl]: Although we don't need a configurable service account for scale tests I updated the code to support the change you've suggested. The original reasoning was about having a hard-coded  static service account for all scale-test applications and create this account in `install.sh` script. But now it's templated as per your suggestion.

File: scale-tests/templates/sort-application.tmpl (hunk 3 above)
Comment (akirillov) [reply scale-tests/templates/sort-application.tmpl]: fixed

**Pull request merged**